vhost_user_net/src/lib.rs

```diff
@@ -183,6 +210,11 @@ impl VhostUserBackend for VhostUserNetBackend {
 
     fn update_memory(&mut self, mem: GuestMemoryMmap) -> VhostUserBackendResult<()> {
         self.mem = Some(mem);
+
+        if self.vrings.is_some() && self.run().is_err() {
+            return Err(Error::FailedRunHandler.into());
+        }
+
```

Comment:
Why create the worker thread here? IMO they either should be created in advance, while initializing VhostUserBackend, or even better, when as VHOST_USER_SET_VRING_KICK message is received.

Also, I don't think "run" is a good name for a method that creates a worker thread an returns.

Reply:
start_worker_thread is added in vhost_user_backend to notify the backend to start worker thread at the point when memory is set for backend in set_mem_table. If we start the worker thread too early, mem is None, so the epoll handler could not handle event. That's why we start the thread once memory is update. set_mem_table is earlier than VHOST_USER_SET_VRING_KICK received, so I think it meets your expectation.